metrics-core/src/traits.rs

```diff
@@ -1,47 +1,59 @@
+// Copyright 2019 Twitter, Inc.
+// Licensed under the Apache License, Version 2.0
+// http://www.apache.org/licenses/LICENSE-2.0
+
 use std::any::Any;
 
 use crate::{Instant, SubMetric};
 
 /// Methods common to all metrics.
 pub trait MetricCommon: Send + Sync {
-    /// Get the current metric as a pointer to a
-    /// type implementing `Any`.
+    /// Get the current metric as a pointer to a type implementing `Any`.
     fn as_any(&self) -> Option<&dyn Any> {
         None
     }
 }
 
+// TODO: consider making these generic in terms of primitive type.
+
```

Comment:
In general this seems like a good idea. One problem here is that the traits need to be able to be used as a dyn Trait which means our ability to do restrict things like this is limited without doing a bigger internal redesign. I'm not opposed to this (being generic over types sounds quite nice) but we need to figure out if we want to put in the effort for that.

Reply:
This is a reasonable tradeoff for now. Might be worth the effort once we've actually merged. I'll leave this todo in-place